pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -238,6 +240,11 @@ public List<String> getTableNamesForQuery(String sqlQuery) {
       Set<String> tableNames = RelToPlanNodeConverter.getTableNamesFromRelRoot(relRoot.rel);
       return new ArrayList<>(tableNames);
     } catch (Throwable t) {
+      String errorMessage = t.getMessage();
+      if (errorMessage.contains("Could not find schema for table: ")) {
```

Comment:
This is not robust. We should not rely on message content.

I saw this exception is thrown from `PinotCatalog`, and `getTable()` should return `null` when it cannot be found based on the interface javadoc

Reply:
In that case, the query parser will return below message, it's more confusion about this `Object 'airlineStats1' not found`
```
ProcessingException(errorCode:150, message:SQLParsingError:
java.lang.RuntimeException: Error composing query plan for: select * from airlineStats1 limit 10
	at org.apache.pinot.query.QueryEnvironment.getTableNamesForQuery(QueryEnvironment.java:248)
	at org.apache.pinot.controller.api.resources.PinotQueryResource.getMultiStageQueryResponse(PinotQueryResource.java:210)
	at org.apache.pinot.controller.api.resources.PinotQueryResource.executeSqlQuery(PinotQueryResource.java:173)
	at org.apache.pinot.controller.api.resources.PinotQueryResource.handlePostSql(PinotQueryResource.java:121)
...
Caused by: org.apache.calcite.runtime.CalciteContextException: From line 1, column 15 to line 1, column 27: Object 'airlineStats1' not found
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
...
Caused by: org.apache.calcite.sql.validate.SqlValidatorException: Object 'airlineStats1' not found
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490))`
```

Reply:
Another thought is to just do: https://github.com/apache/pinot/pull/11330

The whole query parsing here is just to fetch table names. If we use another getTableNames method, then this problem is already gone.